Tests/JSONAPILinksTests.swift

```diff
@@ -0,0 +1,194 @@
+                expect(links.count).toNot(equal(0))
+                expect(links["self"]).to(equal("selfish"))
+            }
+            
+            it("should serialize top level links") {
+                let object = json(JSONAPISerializer(dog, topLinks: ["test": JSONAPILink.Simple(value: "hello"), "another": JSONAPILink.Object(href: "http://example.com/articles/1/comments", meta: Meta())]))
+                let links = object["links"].dictionaryValue
+                expect(links.count).toNot(equal(0))
+                expect(links["test"]).to(equal("hello"))
+                expect(links["another"]!["href"]).to(equal("http://example.com/articles/1/comments"))
+                expect(links["another"]!["meta"]["authors"][0]).to(equal("Yehuda Katz"))
+            }
+        }
+        
+        describe("JSON API Entity links serialization") {
+            let cats = [Cat(id: "33",
+                            name: "Stancho",
+                            links: nil,
+                            topLinks: ["prev": JSONAPILink.Simple(value: "hello"),
+                                       "next": JSONAPILink.Simple(value: "world")]),
+                        Cat(id: "44",
+                            name: "Hez",
+                            links: ["test": JSONAPILink.Simple(value: "hello"),
+                                    "another": JSONAPILink.Object(href: "http://example.com/articles/1/comments", meta: Meta())],
+                            topLinks: ["first": JSONAPILink.Simple(value: "yeah"),
+                                       "last": JSONAPILink.Simple(value: "text")])]
+            let dog = Dog(id: "22",
+                          name: "Joan",
+                          cat: cats[0],
+                          links: nil,
+                          topLinks: ["testDog": JSONAPILink.Simple(value: "hello"),
+                                  "anotherDog": JSONAPILink.Object(href: "http://example.com/articles/1/comments", meta: Meta())])
+            let user = User(id: "11",
+                            name: "Alex",
+                            dog: dog,
+                            cats: cats,
+                            links: ["one": JSONAPILink.Simple(value: "hello"),
+                                    "two": JSONAPILink.Object(href: "hello", meta: Meta())])
+            
+            let user2 = User(id: "39",
+                            name: "Joro",
+                            dog: dog,
+                            cats: cats,
+                            links: ["joroLinkOne": JSONAPILink.Simple(value: "hello"),
+                                    "joroLinkTwo": JSONAPILink.Object(href: "hello", meta: Meta())])
+            
+            it("should serialize the links inside the top data object") {
+                let object = json(user)
+                let links = object["links"].dictionaryValue
+                expect(links.count).toNot(equal(0))
```

Comment:
Instead of testing the count in every test let's create the needed unit tests too check the count (e.g all topLInks are included, all links with meta, etc) and then forget about it